### PR TITLE
`Lectures`: Automatically set name for attachments

### PR DESCRIPTION
**Pull request by krusche**
It is cumbersome to add the title manually if it should be similar to the file name, therefore Artemis now adds it automatically for file attachments and attachment units in case it is not yet specified.
This PR also improves the involved Typescript types to avoid `any`.

#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Create a lecture
2. Add a lecture attachment by choosing a file
3. The name should be filled automatically without the file extension
4. Choose a different file --> the name should not change
5. Add an attachment unit by choosing a file
6. The name should be filled automatically without the file extension
7. Choose a different file --> the name should not change

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

**Review by Strohgelaender — DISMISSED** [review]
code lgtm

**Review by dearjasmina — APPROVED** [review]
works as intended, tested on ts1
is there a reason why we don't update the name each time a new file is selected?

**Review by DominikRemo — APPROVED** [review]
Tested on ts1. Works as expected

Optional:
I get the idea behind not updating the name once it has been set. Otherwise, filling out the name first and then uploading the file would result in the name being overwritten.
However, maybe we could update the name if it was set to the file name before. Something like this:
- If the name is empty -> set it to the filename.
- If the name is not empty, compare it to the previous filename.
    - If there is no previous file upload -> do not update the name.
    - If the name is not equal to the previous filename -> do not update the name.
    - If the name is equal to the previous filename -> set it to the new filename.

File: src/main/java/de/tum/in/www1/artemis/service/notifications/MailService.java [hunk 1]
```diff
@@ -168,14 +163,13 @@ public void sendSAML2SetPasswordMail(User user) {
...
      * @param notificationSubject which has to be a Post
      * @param locale              used for translations
      * @return the modified subject of the email
      */
-    private String setPostContextAndSubject(Context context, Object notificationSubject, Locale locale) {
-        // For Announcement Posts
-        String newAnnouncementString = locale.toString().equals("en") ? newAnnouncementEN : newAnnouncementDE;
+    private String createAnnouncementText(Object notificationSubject, Locale locale) {
+        // Translation that can not be done via i18n Resource Bundle (for Thymeleaf) but has to be set in this service via Java
+        String newAnnouncementString = locale.toString().equals("en") ? "New announcement \"%s\" in course \"%s\"" : "Neue Ankündigung \"%s\" im Kurs \"%s\"";
```
Comment (MaximilianAnzinger) [inline src/main/java/de/tum/in/www1/artemis/service/notifications/MailService.java:172]:
```suggestion
        String newAnnouncementString = "en".equals(locale.toString()) ? "New announcement \"%s\" in course \"%s\"" : "Neue Ankündigung \"%s\" im Kurs \"%s\"";
```
Codacy Static Code Analysis warning

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/attachment-unit-form/attachment-unit-form.component.ts [hunk 2]
```diff
@@ -91,13 +91,21 @@ export class AttachmentUnitFormComponent implements OnInit, OnChanges {
...
+        if (!input.files?.length) {
+            return;
+        }
+        this.file = input.files[0];
+        this.fileName = this.file.name;
+        // automatically set the name in case it is not yet specified
+        if (this.form && (this.nameControl?.value == undefined || this.nameControl?.value == '')) {
+            this.form.patchValue({
+                // without extension
+                name: this.file.name.replace(/\.[^/.]+$/, ''),
```
Comment (MaximilianAnzinger) [inline src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/attachment-unit-form/attachment-unit-form.component.ts:105]: Just a suggestion: we could also replace `_` by spaces
```suggestion
                name: this.file.name.replace(/\.[^/.]+$/, '').replace(/_/g, ' ').trim(),
```

**Review by MaximilianAnzinger — APPROVED** [review]
Tested on TS4. Works as described.

Left two comments in the code.

**Review by RY997 — APPROVED** [review]
Tested on ts3, worked as expected.
Have the same doubt as @dearjasmina

**Review by tobias-lippert — APPROVED** [review]
Tested on ts4. Name is always set automatically as described in the testing steps.

**Pull request merged**